crates/web5/build.rs

```diff
@@ -0,0 +1,18 @@
+use std::process::Command;
+
+fn main() {
+    // Execute the `git rev-parse HEAD` command to get the current commit hash
+    let output = Command::new("git")
+        .args(&["rev-parse", "HEAD"])
+        .output()
+        .expect("Failed to execute git command");
+
+    // Convert the output to a string
+    let git_hash = String::from_utf8(output.stdout).expect("Invalid UTF-8 sequence");
+
+    // Remove the newline character from the commit hash
+    let git_hash_trimmed = git_hash.trim();
+
+    // Pass the commit hash to the compiler as an environment variable
+    println!("cargo:rustc-env=WEB5_GIT_COMMIT_HASH={}", git_hash_trimmed);
```

Comment:
ayy lfg I wonder if this works all the way through to the kt code just like this

I forgot rust has great build script features

Reply:
i'd be surprised if it works in KT, so we might need to set up something separate in the KT logger. i don't know as much about KT compilation flags, so that might take me a hot minute

Reply:
Updated to add the commit hash in kt logs as well

Reply:
so instead, what if in this `build.rs` you added a line of source code inside `lib.rs` that uses the `log_dbg` below to print the git commit hash? that way it's inside the byte code itself such that each sdk we bind to doesn't need to re-implement

Reply:
come to think of it, this may be a good use case for a procedural macro; write a proc macro, maybe call it `log_git_sha` and then add it to `lib.rs` (`log_git_sha!()`) and it is automatically code-gen's at compile time

Reply:
TL;DR it is much simpler to write a separate logging function in Kotlin.

I wasn't able to find a use case for proc macros other than log_dbg. We already have access to the git SHA at compile time via the compile flag `rustc-env=WEB5_GIT_COMMIT_HASH`.

Also, as far as I'm aware, it's not possible to pass a rust macro through UniFFI and call it in the bound language -- in this case, Kotlin. A macro cannot be invoked at runtime, which is necessary for calling it as a bound function in Kotlin. The rust compiler cannot codegen via the macro during Kotlin's compiletime. Also, `log_dbg` is polymorphic, it cannot be described in UDL even if it could be called at runtime.

Another alternative I explored is passing the git SHA through UniFFI. Unfortunately, this approach also came up short. UniFFI does not support passing constants through the language bindings.

This leaves us with two options. First, we can implement a separate logging function in Kotlin. I prefer this. Second, we implement `log_dbg()` as a function, not a macro. Implementing `log_dbg` as a function allows us to pass it across through the language bindings.

Reply:
Like this https://github.com/TBD54566975/web5-rs/pull/290

Lmk whatcha think! I envision more procedural macros in our future so some of this is foundation laying. I think it's slightly preferable because it reduces the overhead for additional bindings and pushes more to the core layer.